lb2/task7/lb2task7.c

```diff
@@ -0,0 +1,102 @@
+#include <stdio.h>
+#include <math.h>
+
+typedef enum {
+    NO_ERROR,
+    ERROR_INVALID_INTERVAL,
+    ERROR_FUNCTION_VALUE_SAME_SIGN
+} ErrorCode;
+
+typedef double (*Function)(double x);
+
+ErrorCode bisection(Function func, double a, double b, double epsilon, double* root) {
+    if (a >= b) {
+        return ERROR_INVALID_INTERVAL;
+    }
+
+    if (func(a) * func(b) > 0) {
+        return ERROR_FUNCTION_VALUE_SAME_SIGN;
+    }
+
+    while ((b - a) / 2 > epsilon) {
```

Comment:
Зачем здесь делить длину отрезка пополам, в определении дихотомии этого правила нет
```suggestion
    while ((b - a) > epsilon) {
```